src/imageio/imageio.c

```diff
@@ -90,6 +90,307 @@
+  { DT_FILETYPE_RGBE, TRUE, 0, 11, dt_imageio_open_rgbe,
+    { '#', '?', 'R', 'A', 'D', 'I', 'A', 'N', 'C', 'E', 0x0A } },
+  // original v1 CRW
+  { DT_FILETYPE_CRW, TRUE, 0, 14, dt_imageio_open_rawspeed,
+    { 'I', 'I', 0x1A, 0x00, 0x00, 0x00, 'H', 'E', 'A', 'P', 'C', 'C', 'D', 'R' } },
+  // most CR2
+  { DT_FILETYPE_CR2, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'I', 'I', '*', 0x00, 0x10, 0x00, 0x00, 0x00, 'C', 'R' } },
+  // CR3 (ISOBMFF)
+  { DT_FILETYPE_CR3, TRUE, 0, 24, dt_imageio_open_libraw,
+    { 0x00, 0x00, 0x00, 0x18, 'f', 't', 'y', 'p', 'c', 'r', 'x', ' ',
+      0x00, 0x00, 0x00, 0x01, 'c', 'r', 'x', ' ', 'i', 's', 'o', 'm' } },
+  // older Canon RAW formats using TIF extension
+  { DT_FILETYPE_CRW, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'I', 'I', '*', 0x00, 0x00, 0x03, 0x00, 0x00, 0xFF, 0x01 } }, // i.e. DCS1
+  { DT_FILETYPE_CRW, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'M', 'M', 0x00, '*', 0x00, 0x00, 0x00, 0x10, 0xBA, 0xB0 } }, // i.e. 1D, 1Ds
+  { DT_FILETYPE_CRW, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'M', 'M', 0x00, '*', 0x00, 0x00, 0x11, 0x34, 0x00, 0x04 } }, // i.e. D2000
+  // older Kodak RAW formats using TIF extension
+  { DT_FILETYPE_KODAK, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'I', 'I', '*', 0x00, 0x00, 0x03, 0x00, 0x00, 0x7C, 0x01 } }, // i.e. DCS460D
+  { DT_FILETYPE_KODAK, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'M', 'M', 0x00, '*', 0x00, 0x00, 0x11, 0xA8, 0x00, 0x04 } }, // i.e. DCS520C
+  { DT_FILETYPE_KODAK, TRUE, 0, 10, dt_imageio_open_rawspeed,
+    { 'M', 'M', 0x00, '*', 0x00, 0x00, 0x11, 0x76, 0x00, 0x04 } }, // i.e. DCS560C
+  // IIQ raw images, may use either .IIQ or .TIF extension
+  { DT_FILETYPE_IIQ, TRUE, 8, 4, dt_imageio_open_rawspeed,
+    { 'I', 'I', 'I', 'I' } },
+  // Fujifilm RAF
+  { DT_FILETYPE_RAF, TRUE, 0, 15, dt_imageio_open_rawspeed,
+    { 'F', 'U', 'J', 'I', 'F', 'I', 'L', 'M', 'C', 'C', 'D', '-', 'R', 'A', 'W' }},
+  // Minolta MRW file
+  { DT_FILETYPE_MRW, TRUE, 0, 4, dt_imageio_open_rawspeed,
+    { 0x00, 'M', 'R', 'M' } },
+  // Olympus ORF file
+  { DT_FILETYPE_ORF, TRUE, 0, 4, dt_imageio_open_rawspeed,
+    { 'I', 'I', 'R', 'O' } },   // most Olympus models
+  { DT_FILETYPE_ORF, TRUE, 0, 4, dt_imageio_open_rawspeed,
+    { 'I', 'I', 'R', 'S' } },	// C7070WZ
+  { DT_FILETYPE_ORF, TRUE, 0, 4, dt_imageio_open_rawspeed,
+    { 'M', 'M', 'O', 'R' } },   // E-10
+  // Panasonic RW2 file
+  { DT_FILETYPE_RW2, TRUE, 0, 8, dt_imageio_open_rawspeed,
+    { 'I', 'I', 'U', 0x00, 0x08, 0x00, 0x00, 0x00 } },
+  // Sigma Foveon X3F file
+  { DT_FILETYPE_X3F, TRUE, 0, 4, NULL,
+    { 'F', 'O', 'V', 'b' } },
+  // little-endian (Intel) TIFF
+  { DT_FILETYPE_TIFF, FALSE, 0, 4, NULL, // may be DNG or any of many camera raw types
+    { 'I', 'I', '*', 0x00 } },
+  // big-endian (Motorola) TIFF
+  { DT_FILETYPE_TIFF, FALSE, 0, 4, NULL, // may be DNG or any of many camera raw types
+    { 'M', 'M', 0x00, '*' } },
+  // little-endian (Intel) BigTIFF
```

Comment:
Just mentioning. According to DNG 1.6 standard These could also be DNG files so not safe for telling the loader properly.

Reply:
That's why I have NULL for the loader function - it will be handled afterwards by trying open_tiff, then open_rawspeed, etc.